Remove fedora-messaging related codes in greenwave/consumers

chuazhan opened the pull request:
As bodhi use ResultsDB and WaiverDB consumers instead of Greenwave
consumer, removing fedora-messaging related codes in
greenwave/consumers.

JIRA: RHELWF-4782

chuazhan commented:
> It seems that still deploy some consumers in Fedora: https://pagure.io/fedora-infra/ansible/blob/main/f/roles/openshift-apps/greenwave/templates/deploymentconfig.yml
> 
> But I think it's the one in `greenwave.consumers.fedora_messaging_consumer` so we can probably remove most of the stuff in `greenwave/consumers` (we do not need it internally anymore).

Ok. I tried to remove all staff in greenwave/consumers, but seems fedora_messaging_consumer is still using ResultsDBHandler and WaiverDBHandler, so I only removed fedmsg related codes.

hluk commented:
Hmm, I'm not sure why this did not fail on the test coverage - Coveralls thinks it improved even though a lot of tests were removed (but not the source files they test). :thinking: 

Can we keep the tests since the related source files where not removed?

chuazhan commented:
> Hmm, I'm not sure why this did not fail on the test coverage - Coveralls thinks it improved even though a lot of tests were removed (but not the source files they test). 🤔

Maybe because I removed the `if self.flask_app.config['MESSAGING'] == 'fedmsg':`, this branch will not be tested again in test coverage.

> Can we keep the tests since the related source files where not removed?
I was thinking those test cases are all for fedmsg, so I remove them totally, now I teak them to fedora-messaging test cases.

chuazhan commented:
I make formatting with black as well, is that ok? It seems too many formatting changes.

hluk commented:
Coveralls (https://coveralls.io/jobs/101790721) shows seemingly big change:

![image](https://user-images.githubusercontent.com/38054/175231511-30e9b3a5-f95e-4055-84aa-fde25540c22f.png)

hluk commented:
> I make formatting with black as well, is that ok? It seems too many formatting changes.

Please change here only the code related to this PR. Can you create a separate commit for it? Greenwave code base is not checked by black (it would be ok to enable this).

hluk commented:
> > Coveralls (https://coveralls.io/jobs/101790721) shows seemingly big change:
> > ![image](https://user-images.githubusercontent.com/38054/175231511-30e9b3a5-f95e-4055-84aa-fde25540c22f.png)
> 
> I don't understand why it downgrade, looks most are old codes.

The removed tests do not test fedmsg specifically, just that the messaging works. It probably tests more cases when decisions differ. Let's try to keep the old tests but use them for fedora-messaging where possible.

chuazhan commented:
> > > Coveralls (https://coveralls.io/jobs/101790721) shows seemingly big change:
> > > ![image](https://user-images.githubusercontent.com/38054/175231511-30e9b3a5-f95e-4055-84aa-fde25540c22f.png)
> > 
> > 
> > I don't understand why it downgrade, looks most are old codes.
> 
> The removed tests do not test fedmsg specifically, just that the messaging works. It probably tests more cases when decisions differ. Let's try to keep the old tests but use them for fedora-messaging where possible.

The old data is 55.97 and now 55.1

hluk commented:
> The old data is 55.97 and now 55.1

OK, that sounds good.

Checking the line coverage and some important lines are not tested but these were not tested even before. :(

Other stuff to **remove**:

- importing fedmsg and using fedmsg everywhere (`fedmsg.consumers.FedmsgConsumer` is unnecessary base class of `Consumer`)
- `except ImportError` when importing `fedora_messaging` since it is now required by the consumers
- Consumer.validate() method
- "MESSAGING" config since there is only one choice now

chuazhan commented:
For `except ImportError when importing fedora_messaging since it is now required by the consumers`, I didn't find where to add this exception. 
BTW, I updated some wording as well.

hluk commented:
> For `except ImportError when importing fedora_messaging since it is now required by the consumers`, I didn't find where to add this exception. BTW, I updated some wording as well.

Do not add it, remove it (from `consumer.py`) - fedora-messaging is now required in consumers so there is no need to import the module *optionally*.

chuazhan commented:
> > For `except ImportError when importing fedora_messaging since it is now required by the consumers`, I didn't find where to add this exception. BTW, I updated some wording as well.
> 
> Do not add it, remove it (from `consumer.py`) - fedora-messaging is now required in consumers so there is no need to import the module _optionally_.

Oh. I misunderstood you :D 
Updated

hluk commented on greenwave/consumers/consumer.py:
```diff
@@ -226,7 +194,5 @@ def _publish_decision_change(
-            if self.flask_app.config['MESSAGING'] == 'fedmsg':
-                self._publish_decision_update_fedmsg(decision)
-            elif self.flask_app.config['MESSAGING'] == 'fedora-messaging':
+            if self.flask_app.config['MESSAGING'] == 'fedora-messaging':
```
Can we drop the `MESSAGING` option completely?

chuazhan replied:
ok. we only have one option now, we don't need to think about whether we will have more options in the future. fixed

hluk commented on greenwave/consumers/consumer.py:
```diff
@@ -59,7 +53,7 @@ def _is_decision_unchanged(old_decision, decision):
 
 
-class Consumer(fedmsg.consumers.FedmsgConsumer):
+class Consumer():
```
```suggestion
class Consumer:
```

chuazhan replied:
Fixed.

hluk commented on greenwave/tests/test_resultsdb_consumer.py:
```diff
@@ -257,7 +251,7 @@ def test_remote_rule_decision_change_not_matching(
     gating.yaml.
     """
     with mock.patch('greenwave.config.Config.MESSAGING', config):
```
You can remove these lines from tests.

chuazhan replied:
Ack. Fixed

hluk commented on greenwave/config.py:
```diff
@@ -45,7 +45,7 @@ class Config(object):
     SUBJECT_TYPES_DIR = '/etc/greenwave/subject_types'
 
-    MESSAGING = 'fedmsg'
+    MESSAGING = 'fedora-messaging'
 
```
```suggestion
```

chuazhan replied:
Ack. Fixed

hluk commented on greenwave/tests/test_resultsdb_consumer.py:
```diff
@@ -143,7 +143,6 @@ def test_no_announcement_subjects_for_old_compose_message():
 
 
 parameters = [
```
You can remove this too.

chuazhan replied:
Ack. Fixed

hluk reviewed (commented):
Can you also remove the Draft flag from the PR?

chuazhan commented on functional-tests/consumers/test_waiverdb.py:
```diff
@@ -60,11 +60,15 @@ def test_consume_new_waiver(
         }
     }
     handler = create_waiverdb_handler(greenwave_server)
+    handler.flask_app.config['MESSAGING'] = 'fedora-messaging'
```
Ack. Fixed
